annotated.go

```diff
@@ -519,16 +942,43 @@ func OnStart(onStart interface{}) Annotation {
 //	   func(lifecycle fx.Lifecycle, p Params) Server {
 //	     server := NewServer(p)
 //	     lifecycle.Append(fx.Hook{
-//		      OnStart: func(ctx context.Context) error {
+//		      OnStop: func(ctx context.Context) error {
 //			    return server.Shutdown(ctx)
 //		      },
 //	     })
+//		 return server
 //	   }
 //	 )
 //
+// It is also possible to use OnStop annotation with other parameter and result
+// annotations, provided that the parameter of the function passed to OnStop
+// matches annotated parameters and results.
+//
+// For example, the following is possible:
+//
+//	fx.Provide(
+//		fx.Annotate(
+//			func (a A) B {...},
+//			fx.ParamTags(`name:"A"`),
+//			fx.ResultTags(`name:"B"`),
+//			fx.OnStop(func (p OnStopParams) {...}),
+//		),
+//	)
+//
+// As long as OnStopParams looks like the following and has no other dependencies
+// besides Context or Lifecycle:
+//
+//	type OnStopParams struct {
+//		fx.In
+//		FieldA A `name:"A"`
+//		FieldB B `name:"B"`
+//	}
+//
 // Only one OnStop annotation may be applied to a given function at a time,
 // however functions may be annotated with other types of lifecylce Hooks, such
-// as OnStart.
+// as OnStop. The hook function passed into OnStop cannot take any arguments
```

Comment:
Misspelled `lifecylce`.

Also this diff replaces `OnStart` with `OnStop` so `OnStop` is used twice in the sentence. The words "other types" makes me think it should be `OnStop` followed by `OnStart` as before this diff.

"Only one **OnStop** annotation ... however functions may be annotated with _other types_ of lifecylce Hooks, such as **OnStop**"